ludwig/utils/backward_compatibility.py

```diff
@@ -106,6 +135,67 @@ def _upgrade_feature(feature: Dict[str, Any]):
     _traverse_dicts(feature, _upgrade_use_bias)
 
 
+def _upgrade_encoder_decoder_params(feature: Dict[str, Any], input_feature: bool):
+    """
+    This function nests un-nested encoder/decoder parameters to conform with the new config structure for 0.6
+    Args:
+        feature (Dict): Feature to nest encoder/decoder params for.
+        input_feature (Bool): Whether this feature is an input feature or not.
+
+    Returns:
+
+    """
+    if input_feature:
+        encoder = feature.get(ENCODER, {})
+        if isinstance(encoder, str):
+            encoder = {TYPE: encoder}
+            feature[ENCODER] = encoder
+
+        nested_params = []
+        for k, v in feature.items():
+            if k not in INPUT_FEATURE_KEYS:
+                encoder[k] = v
+                nested_params.append(k)
+
+        if ENCODER in feature:
+            feature[ENCODER].update(encoder)
+        else:
+            feature[ENCODER] = encoder
+
+        for k in nested_params:
+            del feature[k]
+        warnings.warn(
+            "Encoder specific parameters should now be nested within a dictionary under the 'encoder' parameter. "
+            "Support for un-nested encoder specific parameters will be removed in v0.7",
+            DeprecationWarning,
+        )
```

Comment:
@connor-mccorm Here's one way around the code duplication:

```
def _upgrade_nested_params(feature: Dict[str, Any], type: str) -> None:
    enc_or_dec = feature.get(type, {})
    if isinstance(enc_or_dec, str):
            encoder = {TYPE: enc_or_dec}
            feature[ENCODER] = encoder

        nested_params = []
        for k, v in feature.items():
            if k not in INPUT_FEATURE_KEYS:
                encoder[k] = v
                nested_params.append(k)

        if ENCODER in feature:
            feature[ENCODER].update(encoder)
        else:
            feature[ENCODER] = encoder

        for k in nested_params:
            del feature[k]
        warnings.warn(
            f"{type} specific parameters should now be nested within a dictionary under the '{type}' parameter. "
            f"Support for un-nested {type} specific parameters will be removed in v0.7",
            DeprecationWarning,
        )

def _upgrade_encoder_decoder_params(feature: Dict[str, Any], input_feature: bool):
    if input_feature:
        _upgrade_nested_params(feature, ENCODER)
   else:
       _upgrade_nested_params(feature, DECODER)
```

This might not be exactly right but something along these lines.

Reply:
Refactored this to remove the duplication. Could you possibly take a look and ok it?